sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -372,24 +405,23 @@ internal class ViewAnnotationManagerImpl(
             }
           }
         }
-        hiddenViewMap[annotation.view]?.let { zIndex ->
-          annotation.view.translationZ = zIndex
-          hiddenViewMap.remove(annotation.view)
+
+        if (unpositionedViews.remove(annotation.view)) {
+          annotation.view.visibility = View.VISIBLE
```

Comment:
checked the code - seems that if user has actually done the following:
1. change view from invisible -> visible.
2. view will be added to `unpositionedViews`.
3. before core callback user has made it visible -> invisible.
4. we will return from `OnDrawListener` then
5. core callback executed - and we will actually make the view visible.

Reply:
Theoretically valid concern - though hard to reproduce and not sure what would be the fix. We don't have a way to detect that user called `invisible = true` while view is already invisible. Also, it seems the very same issue had existed before this PR so I suggest to ignore it for now.

Reply:
Discussed offline that in current implementation fixing that seems impossible.